src/snowcli/cli/snowpark/common.py

```diff
@@ -43,6 +44,119 @@ def print_log_lines(file: TextIO, name, id, logs):
         print(_prefix_line(prefix, log + "\n"), file=file, end="", flush=True)
 
 
+def remove_parameter_names(identifier: str):
+    """
+    Removes parameter names from identifier.
+    Deploy commands for function and procedure requires identifier
+    with parameter names (e.g. `hello(number int, name string)`),
+    but describe command requires only parameter types (e.g. `hello(int, string)`)
+    :param identifier: `hello(number int, name string)`
+    :return: `hello(int, string)`
+    """
+    open_parenthesis_index = identifier.index("(")
+    parameters = identifier[open_parenthesis_index + 1 : -1]
+    if not parameters:
+        return identifier
+    types = [t.strip().split(" ")[1] for t in parameters.split(",")]
+    return f"{identifier[0:open_parenthesis_index]}({', '.join(types)})"
+
+
+def check_if_replace_is_required(
+    object_type: ObjectType,
+    current_state,
+    install_coverage_wrapper: bool,
+    handler: str,
+    return_type: str,
+) -> bool:
+    import logging
+
+    log = logging.getLogger(__name__)
+    resource_json = _convert_resource_details_to_dict(current_state)
+    anaconda_packages = resource_json["packages"]
+    log.info(
+        f"Found {len(anaconda_packages)} defined Anaconda "
+        f"packages in deployed {object_type.value}..."
+    )
+    log.info("Checking if app configuration has changed...")
+    updated_package_list = _get_snowflake_packages_delta(
+        anaconda_packages,
+    )
+
+    if object_type == ObjectType.PROCEDURE:
+        coverage_package = "coverage"
+        if install_coverage_wrapper and coverage_package not in [
+            *anaconda_packages,
+            *updated_package_list,
+        ]:
+            updated_package_list.append(coverage_package)
+
+    if updated_package_list:
+        diff = len(updated_package_list) - len(anaconda_packages)
+        log.info(
+            f"Found difference of {diff} packages. Replacing the {object_type.value}."
+        )
+        return True
+    elif (
+        resource_json["handler"].lower() != handler.lower()
+        or _sql_to_python_return_type_mapper(resource_json["returns"]).lower()
+        != return_type.lower()
+    ):
+        log.info(
+            f"Return type or handler types do not match. Replacing the {object_type.value}."
+        )
+        return True
+    return False
+
+
+def _convert_resource_details_to_dict(function_details: SnowflakeCursor) -> dict:
+    import json
+
+    function_dict = {}
+    json_properties = ["packages", "installed_packages"]
+    for function in function_details:
+        if function[0] in json_properties:
+            function_dict[function[0]] = json.loads(
+                function[1].replace("'", '"'),
```

Comment:
Hm, maybe it would be worth to introduce a DTO so we heave easier access than `[0]` or `[1]` WDYT? Maybe we can use `DictCursors` when retrieving the results from snowflake?

Reply:
I only moved this code 😢